utils/gyb_syntax_support/AttributeNodes.py

```diff
@@ -197,5 +199,122 @@
+    #     generic-where-clause?
+    Node('DifferentiableAttributeArguments', kind='Syntax',
+         description='''
+         The arguments for the `@differentiable` attribute: an optional \
+         differentiation parameter list and associated functions.
+         ''',
+         children=[
+             Child('DiffParams', kind='DifferentiationParamsClause',
+                   is_optional=True),
+             Child('DiffParamsComma', kind='CommaToken', description='''
+                   The comma following the differentiation parameters clause,
+                   if it exists.
+                   ''', is_optional=True),
+             Child('MaybePrimal', kind='DifferentiableAttributeFuncSpecifier',
+                   is_optional=True),
+             Child('MaybeAdjoint', kind='DifferentiableAttributeFuncSpecifier',
+                   is_optional=True),
+             Child('MaybeJVP', kind='DifferentiableAttributeFuncSpecifier',
+                   is_optional=True),
+             Child('MaybeVJP', kind='DifferentiableAttributeFuncSpecifier',
+                   is_optional=True),
+             Child('WhereClause', kind='GenericWhereClause', is_optional=True),
+         ]),
+
+    # differentiation-params-clause ->
+    #     'wrt' ':' (differentiation-param | differentiation-params)
+    Node('DifferentiationParamsClause', kind='Syntax',
+         description='A clause containing differentiation parameters.',
+         children=[
+             Child('WrtLabel', kind='IdentifierToken',
+                   text_choices=['wrt'], description='The "wrt" label.'),
+             Child('Colon', kind='ColonToken', description='''
+                   The colon separating "wrt" and the parameter list.
+                   '''),
+             Child('Parameters', kind='Syntax',
+                   node_choices=[
+                       Child('Parameter', kind='DifferentiationParam'),
+                       Child('ParameterList', kind='DifferentiationParams'),
+                   ]),
+         ]),
+
+    # differentiation-params -> '(' differentiation-param-list ')'
+    Node('DifferentiationParams', kind='Syntax',
+         description='The differentiation parameters.',
+         children=[
+             Child('LeftParen', kind='LeftParenToken'),
+             Child('DiffParams', kind='DifferentiationParamList',
+                   collection_element_name='DifferentiationParam',
+                   description='The parameters for differentiation.'),
+             Child('RightParen', kind='RightParenToken'),
+         ]),
+
+    # differentiation-param-list ->
+    #     differentiation-param differentiation-param-list?
+    Node('DifferentiationParamList', kind='SyntaxCollection',
+         element='DifferentiationParam'),
+
+    # differentiation-param -> ('self' | identifer) ','?
+    Node('DifferentiationParam', kind='Syntax',
+         description='''
+         A differentiation parameter: either the "self" identifier or a \
```

Comment:
This description string should be fixed. Currently, it looks like this in [`Sources/SwiftSyntax/gyb_generated/syntax_nodes/SyntaxNodes.swift`](https://github.com/apple/swift-syntax/pull/178/files#diff-be6fd49b9da56aa4d28b732c3f377cd8R7467):

```swift
// MARK: - DifferentiationParamSyntax
/// 
/// A differentiation parameter: either the "self" identifier or a          function parameter name.
/// 
public struct DifferentiationParamSyntax: SyntaxProtocol, SyntaxHashable {
  ...
```
A corresponding change should be made to [apple/swift-syntax](https://github.com/apple/swift-syntax) to update gyb-generated sources, e.g. https://github.com/apple/swift-syntax/pull/178.